src/mpid/ch4/src/ch4_wait.h

```diff
@@ -14,12 +14,44 @@ MPL_STATIC_INLINE_PREFIX int get_vci_wrapper(MPIR_Request * req)
     int vci;
     if (req->kind == MPIR_REQUEST_KIND__PREQUEST_RECV ||
         req->kind == MPIR_REQUEST_KIND__PREQUEST_SEND) {
-        if (req->u.persist.real_request) {
-            vci = MPIDI_Request_get_vci(req->u.persist.real_request);
+        MPIR_Request *real_req = req->u.persist.real_request;
```

Comment:
I know this "real" request assumption is baked into the code right now, but if we could provide a way avoid it, we might be able to see some improvements in the persistent path. We'd need a way preset the VCI at persistent initialization time vs. when the communication is started.

This is orthogonal to this PR, but its something I've been thinking of for a while.